### DISCUSSION TITLE
"Save" icon in toolbar

### DISCUSSION THREAD

**Discussion by myssv** [sumatrapdfreader/sumatrapdf#4288] [Ideas]
Hi there,

i often save my pdf files to an import folder of my paperless dms system.

It would be very nice to have also a save icon in the toolbar!

Thanks for your work

Volker

**Comment by kjk** [comment]
I recommend using Command Palette: `Ctrl + K`, `save file as` (`sav` is probably enough to select this command).

Toolbar doesn't scale: different people want to add different things but if I add all of them, then it'll become overwhelming and unusable for everyone.

**Comment by myssv** [comment]
I can totally understand you, but open, save & print are the main buttons in every program.

Open could also be removed and say the user should press CTRL + O  :-)

Don't get me wrong, it was youst an idea and maybe the other usere will push it.

We will see!

Have a great day!

**Comment by kjk** [comment]
You can now add custom buttons to the toolbar. See https://www.sumatrapdfreader.org/docs/Customize-toolbar

You can add toolbar for `CmdSaveAs` command.

**Reply by inglorious4** [reply]
This works nicely in the latest preview release.  I added the CmdSaveAs button (tool text Save As) to my toolbar easily.

**Comment by dominiosc** [comment]
Is possible remove "..." or set the size of button? when use a word o a letter it's cut.

with word "Save"
![imagen](https://github.com/user-attachments/assets/e87fa0ba-bfaf-4b4b-8f54-d46926bcfd55)

with letter "S"
![imagen](https://github.com/user-attachments/assets/5426194b-3731-4619-95c8-be0711dc7e96)

**Reply by GitHubRulesOK** [reply]
@dominiosc currently not easy this was requested but stated as not easy to alter as tied to other UI scaling 

You can make a slight change using settings so here I have Manga as an option for comic books but although the command works for PDF it is **non- active** when a PDF is open
![image](https://github.com/user-attachments/assets/b231446e-3aa1-4381-843e-f34380561634)
![image](https://github.com/user-attachments/assets/b18add7f-40f2-496a-a7ea-ecacd2c9116e)
![image](https://github.com/user-attachments/assets/5c17907a-06f7-401b-83ff-d74263b2e7d3)

**Reply by HeroMickeyYu** [reply]
Hi, can the CmdToggleMangaMode command take effect in PDF files now? This is how I set it up,

```
Shortcuts [
	[
		Cmd = CmdToggleMangaMode
		Key = Ctrl + M
		ToolbarText = Manga
	]
]
```

Just like in the picture, when I open a PDF file, the manga toolbar does not work

**Reply by GitHubRulesOK** [reply]
it should still work as a keyboard function in PDF no problem but I seem to recollect it has not always been accepted for PDF an in more recent versions as stated is non active (greyed out) for PDF !

**Reply by dominiosc** [reply]
@GitHubRulesOK How do you display the whole word? In my case, it always shows one character and three dots.
Could it be due to some advanced configuration?

SidebarDx = 523
ToolbarSize = 42

<img width="1054" height="56" alt="image" src="https://github.com/user-attachments/assets/5fa2d92a-dc77-48b2-a5c2-1be8ed96ad8d" />

**Reply by GitHubRulesOK** [reply]
@dominiosc 
the text size must be under 20 to see whole line (e.g. this is 19)
<img width="245" height="42" alt="image" src="https://github.com/user-attachments/assets/96f7d40f-5e57-4ee9-be5b-b26c8bad7a1d" />

**Comment by marconedev** [comment]
You can create a shortcut save icon with this code:

```
Shortcuts [
	[
		Cmd = CmdSaveAnnotations
		ToolbarText = 💾
	]
]
```

**Comment by JannetGen** [comment]
I feel like Sumatra is so fast that I often forget how many cool customization options are actually hidden under the hood. Adding a more direct way to toggle this would definitely save a few trips to the manual.
